### PR TITLE
Strip leading 'v' from git tags if present.

### PR DESCRIPTION
**Pull request by wadells**
As discussed at:

  https://github.com/golang/go/issues/32945
  https://semver.org/#is-v123-a-semantic-version

Go and semver disagree about handling of a leading 'v' on version
strings.  This patch allows robotest to play nicely with both.

I discovered this when I created the v2.0.0-alpha.1 tag in git.

Contributes to https://github.com/gravitational/robotest/issues/200.

### Testing Done
Before:
```
walt@work:~/git/robotest$ git describe --tags
v2.0.0-alpha.1
walt@work:~/git/robotest$ make version       
version metadata saved to version.go
Robotest Version: v2.0.0-alpha.1
```
After:
```
walt@work:~/git/robotest$ git describe --tags        
v2.0.0-alpha.1-1-g4ac1444
walt@work:~/git/robotest$ make version       
version metadata saved to version.go
Robotest Version: 2.0.0-alpha.1.1+4ac14441
```
With a non-v-prefixed tag:
```
walt@work:~/git/robotest$ git tag 3.fake.version
walt@work:~/git/robotest$ make version
version metadata saved to version.go
Robotest Version: 3.fake.version
walt@work:~/git/robotest$ git tag -d 3.fake.version 
Deleted tag '3.fake.version' (was 4ac1444)
```

### CODE REVIEW

File: version.sh [hunk 1]
```diff
@@ -29,11 +29,20 @@ COMMITS_SINCE_LAST_TAG=`git rev-list  ${COMMIT_WITH_LAST_TAG}..HEAD --count`
 BUILD_METADATA=`git rev-parse --short=8 HEAD`
 DIRTY_AFFIX=$(git diff --quiet || echo '-dirty')
 
+# strip leading v from git tag, see:
```
Comment (a-palchikov) [inline version.sh:32]: Header comment should be updated to reflect the change.
On a side note - what benefit does this bring vs tagging without the `v`?

Comment (wadells) [reply version.sh:32]: >  On a side note - what benefit does this bring vs tagging without the v?

TLDR: `v` prefixes are what the wider go ecosystem use

To quote the go developers:

> The difference is mainly aesthetic and relatively arbitrary. Absent a compelling need otherwise, Go usually resolves aesthetic differences by picking one to use consistently (think of gofmt), not supporting both.

https://github.com/golang/go/issues/32945#issuecomment-512900259

`v` prefixed git tags are the right choice for go libraries because `go mod` reverts to pseudo-versions if there is no `v` in the git tag.  And if we're going to track libraries via a pseudo version, why bother to make git version tags at all?

A quite reasonable counter argument is: YAGNI, robotest will never be used as a library.  However why have  different version tagging strategies for libraries vs applications? "Resolve aesthetic differences by picking one to use consistently"

I know Gravity made a choice to use non-v-prefixed before go modules were widely available, but I doubt we'd make the same choice today for a new go project.  You could argue robotest made the same choice, but two tags (0.0.1 and 1.0.0 -- neither of which is used today) don't carry the same responsibility to the customers of the project as Gravity's longstanding & widely used release tags.

Effectively, I'm treating 2.0.0 as the first semantically versioned release, and everything prior as "dragons be here" jumbled prehistory  -- in a large part because it isn't easy determine which robotest code is running before the recent versioning logging changes (https://github.com/gravitational/robotest/pull/206 and https://github.com/gravitational/robotest/pull/215) that will be available in 2.0.0.

/ramble

Comment (wadells) [reply version.sh:32]: I updated the comments in fb8125a.  Thanks for helping me keep up with documentation.
